core/modules/simpletest/tests/bootstrap.test

```diff
@@ -139,11 +139,14 @@ class BootstrapPageCacheTestCase extends BackdropWebTestCase {
     $this->assertResponse(304, 'Conditional request with obsolete If-Modified-Since date returned 304 Not Modified.');
 
     $this->backdropGet($test_path, array(), array('If-Modified-Since: ' . $last_modified));
-    $this->assertResponse(200, 'Conditional request without If-None-Match returned 200 OK.');
+    $this->assertResponse(304, 'Conditional request without If-None-Match returned 304 Not Modified.');
+
+    $this->backdropGet($test_path, array(), array('If-Modified-Since: ' . gmdate(DATE_RFC1123, strtotime($last_modified) + 1)));
+    $this->assertResponse(200, 'Conditional request with new a If-Modified-Since date newer than Last-Modified returned 200 OK.');
```

Comment:
Same as above: I'd expect 200 with `strtotime($last_modified) - 1`, 304 with `strtotime($last_modified) + 1`.

If the `If-Modified-Since` header provided has a date newer (+1) than the cached resource, it should be 304, because the page didn't change since then. The page I already have is still valid.
If the header provides a date *before* (-1) the date the page was cached, it should be 200. The page I have is outdated.

Reply:
Hm, thanks I think you're right on both comments. Does that mean that if you run this test on your local, it fails? While if we switched this to +1 and inverted the conditional above, it would pass?

Reply:
I have to admit, I didn't run it on my local yet. I'd assume that the test passes, because in both places the logic is switched (compared to what I'd expect).

Reply:
@quicksketch According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-None-Match the If-None-Match header takes priority of the If-Modified-Since header, so the tests will still fail when using mod_proxy. Essentially what we need to do is check If-None-Match on it's own, and then only check If-Modified-Since if the If-None-Match header is not set.